### PR TITLE
Add support for localized font families

### PR DESCRIPTION
**Pull request by marinaaisa**
Bug/issue #108129110, if applicable: 

## Summary

Add support for localized font families

## Dependencies

NA

## Testing

Steps:
1. Make sure current fonts look correct 

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by marinaaisa** [comment]
@swift-ci test

File: src/styles/core/_vars.scss [hunk 1]
```diff
@@ -81,3 +81,11 @@ $on-this-page-aside-width: 192px;
...
 // Suggest Lang
 $suggest-lang-height: 52px;
+
+// Localized Quotes
+$localizedQuotes: (
+  ja-JP: "「" "」",
+);
+
+// Localized Fonts
+$localizedFonts: ()
```
Comment (dobromir-hristov) [inline src/styles/core/_vars.scss:91]: I would make them default + add some comments how to use them. Maybe the structure for the `$localizedFonts`

```suggestion
$localizedQuotes: (
  ja-JP: "「" "」",
) !default;

// Localized Fonts
$localizedFonts: () !default;
```

File: src/styles/base/_typography.scss [hunk 2]
```diff
@@ -19,6 +19,12 @@ html {
     font: var(--typography-html-font, $body-font-size $font-content);
   }
   quotes: "“" "”";
+
+  @each $lang, $quotes in $localizedQuotes {
+    :lang(#{$lang}) {
```
Comment (dobromir-hristov) [inline src/styles/base/_typography.scss:24]: We apply the lang on the `html` right?
```suggestion
    &:lang(#{$lang}) {
```

File: src/styles/base/_typography.scss [hunk 3]
```diff
@@ -19,6 +19,12 @@ html {
     font: var(--typography-html-font, $body-font-size $font-content);
   }
   quotes: "“" "”";
+
+  @each $lang, $quotes in $localizedQuotes {
+    &:lang(#{$lang}) {
+      quotes: $quotes;
```
Comment (mportiz08) [inline src/styles/base/_typography.scss:25]: This property only applies to quotes that are either added through a CSS `content` field or the HTML `q` element, right and normal text content would have to be already localized in the text itself, right?

Comment (marinaaisa) [reply src/styles/base/_typography.scss:25]: Exactly, yes.

File: src/styles/core/_vars.scss (hunk 1 above)
Comment (marinaaisa) [reply src/styles/core/_vars.scss]: https://github.com/apple/swift-docc-render/pull/677/commits/90f0b4ad049fd9087e1cc115574c02c1fd3ac7d9

**Comment by marinaaisa** [comment]
@swift-ci test

**Pull request merged**